sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -121,8 +123,13 @@ class SessionCatalog(
     currentDb = db
   }
 
-  def getDefaultDBPath(db: String): String = {
-    System.getProperty("java.io.tmpdir") + File.separator + db + ".db"
+  def getDatabasePath(dbName: String, path: Option[String]): String = {
+    val dbPath =
+      path.map(new Path(_))
+        .getOrElse(
+          new Path(new Path(System.getProperty("java.io.tmpdir")), dbName.toLowerCase() + ".db"))
+    val dbLocation = dbPath.toString
+    if (dbLocation.endsWith(File.separator)) dbLocation.dropRight(1) else dbLocation
```

Comment:
So far, the code is not used. Basically, this implementation is a dead code, but it still has two issues. 

First, on some platform, the temporary directory `java.io.tmpdir` does not include a trailing slash. Thus, using `new Path` will handle all the issues we might miss. It is consistent with what we did in `HiveSessionCatalog`.

Second, we need to call `toLowerCase()` when generating the path. This is to make it consistent with what we did in `HiveSessionCatalog`. 

So far, we are not creating any physical directory when we creating a database in `SQLContext`. We just store the information in the memory. Basically, this database is a logical concept, which corresponds to a set of tables. We can set it to an empty string, it should be still OK. Let me know if you want me to set it to an empty string. Thanks!